subsys/bluetooth/host/gatt.c

```diff
@@ -657,6 +684,15 @@ ssize_t bt_gatt_attr_write_ccc(struct bt_conn *conn,
 	/* Update cfg if don't match */
 	if (ccc->cfg[i].value != ccc->value) {
 		gatt_ccc_changed(attr, ccc);
+
+#if defined(CONFIG_BT_SETTINGS_CCC_STORE_ON_WRITE)
+		if (atomic_cas(&gatt_ccc_store.scheduled, false, true) &&
+		    bt_addr_le_is_bonded(conn->id, &conn->le.dst)) {
+			gatt_ccc_store.conn = bt_conn_ref(conn);
```

Comment:
How does this work with multiple connections? Wont it overwrite gatt_ccc_store.conn if it was already set for another connection? (and thereby cause a leak). I suppose there needs to be a per-connection storage rather than this single `gatt_ccc_store` for all connections.

Reply:
Yes, it's possible that when one connection scheduled a write, another would schedule its own before the first is executed. What would be the best way to keep it per-connection? Note that we don't need the bt_conn object itself, rather I'd and peer address for `bt_gatt_store_ccc()`

Reply:
@Qbicz You should probably use a list to store the connection which are pending to store, so you actually reuse the same work.